src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -130,12 +130,102 @@ export const currentChannelDisplayableMessages = createSelector(
         id: message.id,
         type: message.type,
         message: message.message,
-        createdAt: formatMessageDisplayDate(message.createdAt),
+        createdAt: message.createdAt,
         nickname: user.username,
       };
     })
 );
 
+// returns array of messages SORTED by createdAt
+export const currentChannelMessagesSortedByTime = createSelector(
+  currentChannelDisplayableMessages,
+  (messages) =>
+    messages.sort((a, b) => b.createdAt - a.createdAt)
+);
+
+// returns array of messages SORTED and GROUPED by createdAt and user
+export const currentChannelMessagesGroupedByUserTime = createSelector(
```

Comment:
Maybe it's just cosmetic change but I'd rather named this selector 'currentChannelMessagesMergedBySender' as it's more precise

Reply:
agree